Mage/Command/BuiltIn/CompileCommand.php

```diff
@@ -22,19 +22,43 @@
 class CompileCommand extends AbstractCommand
 {
     /**
+     * @var Compiler
+     */
+    private $compiler;
+
+    public function __construct()
+    {
+        $this->setCompiler(new Compiler());
+    }
+
```

Comment:
Why not the following constructor code?

``` php
    /**
     * @var Compiler
     */
    private $compiler;

    public function __construct(Compiler $compiler = null)
    {
        if (is_null($compiler)) {
            $compiler = new Compiler();
        }
        
        $this->compiler = $compiler;
    }
```

This way, you don't need the setter and won't allow to change the compiler on runtime like the previews version.